src/components/Forms/Select.svelte

```diff
@@ -0,0 +1,160 @@
+<script>
+  import { themeName } from "../../view";
+  export let value, select, disabled;
+</script>
+
+<style lang="scss">
+  .dropdown {
+    display: inline-block;
+    position: relative;
+    overflow: hidden;
+    height: 2.5em;
+    background: #f2f2f2;
+    border: 1px solid;
+    border-color: white #f7f7f7 whitesmoke;
+    border-radius: 3px;
+    background-image: -webkit-linear-gradient(top, transparent, rgba(0, 0, 0, 0.06));
+    background-image: -moz-linear-gradient(top, transparent, rgba(0, 0, 0, 0.06));
+    background-image: -o-linear-gradient(top, transparent, rgba(0, 0, 0, 0.06));
+    background-image: linear-gradient(to bottom, transparent, rgba(0, 0, 0, 0.06));
+    -webkit-box-shadow: 0 1px 1px rgba(0, 0, 0, 0.08);
+    box-shadow: 0 1px 1px rgba(0, 0, 0, 0.08);
+  }
+
+  .dropdown:before,
+  .dropdown:after {
+    content: "";
+    position: absolute;
+    z-index: 2;
+    top: 0.85em;
+    right: 10px;
+    width: 0;
+    height: 0;
+    border: 4px dashed;
+    border-color: #888888 transparent;
+    pointer-events: none;
+  }
+
+  .dropdown:before {
+    border-bottom-style: solid;
+    border-top: none;
+  }
+
+  .dropdown:after {
+    margin-top: 0.475em;
+    border-top-style: solid;
+    border-bottom: none;
+  }
+
+  .dropdown-select {
+    position: relative;
+    width: 130%;
+    margin: 0;
+    padding: 6px 8px 6px 10px;
+    height: 100%;
+    font-size: 0.9rem;
+    text-shadow: 0 1px white;
+    color: #62717a;
+    background: #f2f2f2; /* Fallback for IE 8 */
+    background: rgba(0, 0, 0, 0) !important; /* "transparent" doesn't work with Opera */
+    border: 0;
+    border-radius: 0;
+    -webkit-appearance: none;
+  }
+
+  .dropdown-select:focus {
+    z-index: 3;
+    width: 100%;
+    outline: 0;
+  }
+  .dropdown:focus-within {
+    border-color: var(--accent);
+    box-shadow: 0px 0px 4px 1px var(--accent);
+    outline: 0;
+  }
+
+  :global(.dropdown-select > option) {
+    margin: 3px;
+    padding: 6px 8px;
+    text-shadow: none;
+    background: #f2f2f2;
+    border-radius: 3px;
+    cursor: pointer;
+  }
+
+  /* Dirty fix for Firefox adding padding where it shouldn't. */
```

Comment:
is there a link for what/why firefox is doing?

Reply:
I don't know, I copied that code